pkg/util/liveconfig/hive.go

```diff
@@ -0,0 +1,54 @@
+package liveconfig
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	HiveKeyvaultSecret = "HiveConfig"
+)
+
+type hiveShard struct {
+	Kubeconfig []byte `json:"kubeconfig,omitempty"`
+}
+
+type hiveConfig struct {
+	Shards []hiveShard `json:"shards,omitempty"`
+}
+
+func (p *prod) HiveRestConfig(ctx context.Context, index int) (*rest.Config, error) {
+	secret, err := p.kv.GetSecret(ctx, HiveKeyvaultSecret)
```

Comment:
is there anyway we can verify we have all required config defined in the keyvault before we fail on grabbing individual components of the  "live config"?